leakcanary-android-instrumentation/src/main/java/leakcanary/FailTestOnLeakRunListener.kt

```diff
@@ -63,8 +63,11 @@ open class FailTestOnLeakRunListener : RunListener() {
    * is started. Returns null to continue leak detection, or a string describing the reason for
    * skipping otherwise.
    */
-  protected open fun skipLeakDetectionReason(description: Description): String? {
-    return null
+  internal open fun skipLeakDetectionReason(description: Description): String? {
+    return if (description.getAnnotation(LeakTest::class.java) != null)
+      null
+    else
+      "skipped leak detection"
```

Comment:
This changes the default behavior of `FailTestOnLeakRunListener`. I'd rather have `FailTestOnLeakRunListener.kt` still behave the same.

One thing you can do is add two new subclasses, e.g.`WhiteListFailTestOnLeakRunListener` (uses `@FailTestOnLeak`) and `BlackListFailTestOnLeakRunListener` (uses `@SkipLeakDetection`)

Reply:
Yeah, just did the change. Makes sense to me, otherwise the name of the `FailTestOnLeakRunListener` is misleading if by default doesn't fail. Having two listeners and let the user choose sounds good to me 👍